tensorflow/java/src/gen/cc/op_gen_main.cc

```diff
@@ -48,8 +48,11 @@ const char kUsageHeader[] =
     "through\n"
     "the 'org.tensorflow.op.Ops' API as a group until the generated classes "
     "are compiled using an appropriate annotation processor.\n\n"
-    "Finally, the '--base_package' overrides the default parent package "
-    "under which the generated subpackage and classes are to be located.\n\n";
+    "The '--base_package' overrides the default parent package under which "
+    "the generated subpackage and classes are to be located.\n\n"
+    "Finally, a list of directories of API proto definitions can be provided "
```

Comment:
That's what I thought but looks like there is no group name in the `ApiDef` neither. I was hoping to find it in the `ApiDef_Endpoint.name()`, as the "Package" like it mentioned in the proto definition here:
>>   // Name should be either like "CamelCaseName" or
    // "Package.CamelCaseName". Client-language-specific ApiDefs may
    // use a snake_case convention instead of CamelCase.
    string name = 1;

but it looks that package is always empty and only the endpoint name part is present. Am I wrong?

Reply:
You're not wrong in that those aren't present yet.
But we should add them and use that mechanism to organize instead of lib names.
The Python API is organized like that with ApiDef overrides, for example: [1](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/python_api/api_def_DecodeJpeg.pbtxt), [2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/python_api/api_def_BatchMatrixTriangularSolve.pbtxt), [3](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/api_def/python_api/api_def_RightShift.pbtxt) etc.